### ISSUE TITLE
Menu options not grayed out when dialog is open, so commands can get buffered and executed

### ISSUE THREAD

**Issue by ronblum** [rstudio/rstudio#2482]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.2.462
    OS Version      : MacOS 10.12.6 (Sierra)
    R Version       : 3.4.4

### Steps to reproduce the problem

Choose a menu option with a dialog, followed by other menu items.

1. `Help -> Check for Updates`. Do _not_ click `OK`
2. `Tools -> Terminal -> New Terminal`
3. `Session -> New Session`
4. `File -> New File -> R Markdown`
5. `Code -> Source`
6. Click `OK`

### Describe the problem in detail

All the actions in steps 2–5 are executed: a new terminal is created, a new session window is spawned, a dialog for a new R Markdown file pops up, and the code that was in the Source pane is sourced.

### Describe the behavior you expected

None of the actions should be executed. The only thing that should happen after the `OK` button is clicked is that the `Check for Updates` dialog should disappear. This is a regression: in 1.1.442, most menu items were grayed out, so this wasn't possible. Even if the menu items aren't grayed out, they shouldn't be buffered and executed.

`Check for Upates` is just one example. Other dialogs—for example, `Save As`—will do the same. Granted, most of the time, people will perform only one action at a time. However, sometimes people will poke around and (sort of) unintentionally get themselves in a possibly bad state.

**Comment by jmcphers** [comment]
Agree that this is a regression and that it shouldn't work this way, but I don't think this usage pattern will be common so moving out of 1.2 for now.
